modules/services/gromit-mpx.nix

```diff
@@ -0,0 +1,212 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.gromit-mpx;
+
+  # Select the appropriate hot key:
+  hotkey =
+    if cfg.hotKeyCode != null
+      then "--keycode ${toString cfg.hotKeyCode}"
+      else if cfg.hotKey != null
+        then "--key ${cfg.hotKey}"
+        else "--key none";
+
+  # Select the appropriate undo key:
+  undokey =
+    if cfg.undoKeyCode != null
+      then "--undo-keycode ${toString cfg.undoKeyCode}"
+      else if cfg.undoKey != null
+        then "--undo-key ${cfg.undoKey}"
+        else "--undo-key none";
+
+  # The command line to send to gromit-mpx:
+  commandArgs = concatStringsSep " " [
+    hotkey
+    undokey
+    "--opacity ${toString cfg.opacity}"
+  ];
+
+  # Allowed modifiers:
+  modsAndButtons = [
+    "1" "2" "3" "4" "5"
+    "SHIFT" "CONTROL" "ALT" "META"
+  ];
+
+  # Create a string of tool attributes:
+  toolAttrs = tool: concatStringsSep " " (
+    [ "size=${toString tool.size}" ] ++
+    optional (tool.type != "eraser") ''color="${tool.color}"'' ++
+    optional (tool.arrowSize != null) "arrowsize=${toString tool.arrowSize}");
```

Comment:
We typically put the operator at the beginning of the lines when spanning multiple lines. So this becomes
```
  toolAttrs = tool: concatStringsSep " " (
    [ "size=${toString tool.size}" ]
    ++ optional (tool.type != "eraser") ''color="${tool.color}"''
    ++ optional (tool.arrowSize != null) "arrowsize=${toString tool.arrowSize}");
```